### ISSUE TITLE
Why do you scale the label when resizing the image?

### ISSUE THREAD

**Issue by Olli-Olli** [YvanYin/Metric3D#140]
First of all, great job you did here! I have one question, something I do not get my head around.

If I get your your paper correctly, you say in section 3.2 that "the ground truth labels are resized without any scaling". When I look into your code I understand you do scale the labels after resizing the image (main/mono/utils/do_test.py lines 97ff). I have images that I just want to resize to fit the network dimensions. You explain well, that this affects the intrinsics. But why would you scale the depth labels using the to_scale_ratio? Applying the to_canonical_ratio should suffice, shouldn't it?

**Comment by alesnick-ai** [comment]
to the best of my understanding -- the transformer or convnet backbone requires a fixed input resolution.  their solution does, indeed, scale the depth predictions rather than scale the image.  however, there still needs to be a rescaling/padding simply to conform the image to the resolution that the neural network is expecting

**Comment by alesnick-ai** [comment]
(and that rescaling must also propagate through the intrinsics)

**Comment by Olli-Olli** [comment]
I understand the image and depth need to be resized to fit the neural network input dimension. My understanding is that the depths can just be resized without having to scale the depth as well. I understand that this scaling is done in the code, though. I might be mistaken...

**Comment by jbrownkramer** [comment]
Let's say you have a picture you want to infer depths on.  You want to fit the whole thing into the network, so you have to resize it.  In doing so, the intrinsics of the image change.  Crucially they change to some random thing, and are not forced to focal length = 1000 pixels.  The network was trained with f  = 1000.  Fundamentally it has to know what a typical scale is for certain objects.  For example, a typical sized person at a meter away might be 250 pixels wide, say.  If the camera intrinsics are f = 500, then that person will be 125 pixels wide.  But the network is unaware of the true focal length, so it will report the person to be 2 meters away.  Hence the need to scale the depths by the ratio of the true focal length / 1000.

**Comment by Olli-Olli** [comment]
This is a very good explanation @jbrownkramer, Thanks.
